subworkflows/local/fusioninspector_workflow.nf

```diff
@@ -17,19 +17,23 @@ workflow FUSIONINSPECTOR_WORKFLOW {
     main:
         ch_versions = Channel.empty()
         index ="${params.starfusion_ref}"
-        ch_fusion_list = params.fusioninspector_filter ? fusion_list_filtered : fusion_list
+        ch_fusion_list = ( params.fusioninspector_filter ? fusion_list_filtered : fusion_list )
+        .branch{
+            no_fusions: it[1].size() == 0
+            fusions: it[1].size() > 0
+        }
 
         if (params.whitelist)  {
-            ch_whitelist = ch_fusion_list.combine(Channel.value(file(params.whitelist, checkIfExists:true)))
+            ch_whitelist = ch_fusion_list.fusions.combine(Channel.value(file(params.whitelist, checkIfExists:true)))
```

Comment:
What happens if there are no fusions here? It creates a ch_fusion_list.no_fusions above, but the here ch_fusion_list.fusions would be NULL, then this code would essentially output `ch_whitelist = null`. Then  `CAT_CAT(ch_whitelist)` should lead to a NullPointerException or a similar error, is that correct?

Then maybe we could check first whether there are fusions?
```
if (ch_whitelist) {
    CAT_CAT(ch_whitelist)
} else {
    // some action when there are no fusions
}
```

Reply:
From what I understand: in case of no fusions, ch_fusion_list.fusions and ch_whitelist do not exist and the processes are not triggered. No error is thrown.

Reply:
Ok awesome 👍